framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -152,6 +166,7 @@ bool things_ping_set_mask(const char *remote_addr, uint16_t port, ping_state_e s
 		return false;
 	}
 
+	pthread_mutex_lock(&mutex_ping_list);
 	if (list == NULL) {
 		THINGS_LOG_V(TAG, "OICPing Module is not initialized.");
 		return false;
```

Comment:
you need to unlock here.

Reply:
The mutex will be unlocked at line no. 187.